applications/conceptual-model-editor/src/app/diagram/reactflow/components/menu-options.tsx

```diff
@@ -43,40 +45,74 @@ export const useMenuOptions = () => {
+     * - handlersWithTexts is an array of tuples. The first property specifies the handler called on click and the second one the name. 
+     * Special case is if the handler is undefined and the name is "HorizontalSeparator".
+     * - positionTailwind specifies the position in tailwind notation.
+     * - positionCSS specifies the position in CSS notation. If both the tailwind and CSS positions are specified, the behavior is undefined.
+     * @returns
+     */
+    const MenuOptionsGeneral = (props: {
+        handlersWithTexts: Array<[(() => void) | undefined, string]>;
+        positionTailwind?: string;
+        positionCSS?: object;       // TODO: It might be possible to specify better type
+    }) => {        
+        const buttonsToRender = props.handlersWithTexts.map(e => {
+            const [handler, text] = e;
+            return ((text === "HorizontalSeparator" && handler === undefined) ? (<HorizontalSeparator />) :
+                (handler === undefined) ? (<></>) : (<MenuButton text={text} onClick={handler} />));                               
+        });              
+        
+        
+        let style: React.CSSProperties = { pointerEvents: "all" };
+        if (props.positionCSS !== undefined) {
+            style = { 
+                ...style, 
+                ...props.positionCSS
+            };
+        }
         return (
             <div
                 ref={menuOptionsRef}
                 tabIndex={-1}
-                style={{ pointerEvents: "all" }}
-                className={`flex flex-col bg-white border-2 border-slate-400 border-solid [&>*]:px-5 [&>*]:text-left ${props.position ? props.position : ""}`}
+                style={style}
+                className={`flex flex-col bg-white border-2 border-slate-400 border-solid [&>*]:px-5 [&>*]:text-left ${props.positionTailwind ? props.positionTailwind : ""}`}                
                 onBlur={(event) => {
                     if (event.relatedTarget?.id.startsWith("button-menu-options-")) {
                         return;
                     }
                     close(event);
                 }}
             >
-                <MenuButton text="ℹ Detail" onClick={openDetailHandler} />
-                {modifyHandler === undefined ? null :
-                    <MenuButton text="✏ Modify" onClick={modifyHandler} />}
-                <HorizontalSeparator />
-                {createProfileHandler === undefined ? null : <>
-                    <MenuButton text="🧲 Create profile" onClick={createProfileHandler} />
-                    <HorizontalSeparator />
-                </>}
-                {removeFromViewHandler === undefined ? null :
-                    <MenuButton text="🕶 Remove from view" onClick={removeFromViewHandler} />}
-                {deleteHandler === undefined ? null :
-                    <MenuButton text="🗑 Delete" onClick={deleteHandler} />}
+                {
+                    buttonsToRender
```

Comment:
I belive this is a false advertising .. it is not only buttons but also separators. Please use true and descriptive naming.